File: ELEC881_Copilot/src/Original Prompt/Java script (JS)/q3-pascals-triangle-ii.js
/*
Given an integer rowIndex, return the rowIndexth (0-indexed) row of the Pascal's triangle.

In Pascal's triangle, each number is the sum of the two numbers directly above it as shown:


 

Example 1:

Input: rowIndex = 3
Output: [1,3,3,1]
Example 2:

Input: rowIndex = 0
Output: [1]
Example 3:

Input: rowIndex = 1
Output: [1,1]
 

Constraints:

0 <= rowIndex <= 33
*/
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    let result = [1];
    for (let i = 1; i <= rowIndex; i++) {
        let next = [];
        next[0] = 1;
        for (let j = 1; j < result.length; j++) {
            next[j] = result[j] + result[j - 1];
        }
        next.push(1);
        result = next;
    }
    return result;
    
};

File: ELEC881_Tabnine/src/No Description Prompt/Python/q3-pascals-triangle-ii.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex < 0:
            return []

        row = [1] * (rowIndex + 1)
        for i in range(1, rowIndex // 2 + 1):
            row[i] = row[-i - 1] = row[i - 1] * (rowIndex - i + 1) // i
        return row
        